hist/histv7/test/constructors.cxx

```diff
@@ -0,0 +1,90 @@
+#include "gtest/gtest.h"
+
+#include "ROOT/RHist.hxx"
+
+// Test RH.. constructors and especially GetNBins() and GetBinContent() from RHistImpl.hxx
+
+// Test RH1F constructor
+TEST(HistConstructorTest, ConstructorRH1F) {
+  ROOT::Experimental::RH1F hist({100, 0., 1});
+  EXPECT_EQ(102, hist.GetImpl()->GetNBins());
+  EXPECT_EQ(0, hist.GetImpl()->GetBinContent(42));
+  hist.GetImpl()->AddBinContent(42,.42f);
+  EXPECT_FLOAT_EQ(.42f, hist.GetImpl()->GetBinContent(42));
```

Comment:
Not super sure if these are best placed in a "constructors" test. I would personally expect to find this sort of tests into a "filling" or "bin manipulation" test.

Also, note that GetBinContent/AddBinContent only exercises the "global" data binning (as reported by GetNBins), which works in the same way no matter what the histogram dimension is. In order to exercise the sort of tricky local <-> global bin conversion that I was referring to in your earlier PR, you will need to also call methods that manipulate local histogram coordinates, such as `Fill` or `RHist::GetBinContent` (or`GetBinIndex` and `GetBin[Center|From|To]` from RHistImpl, but these are already reasonably well covered by the "binning" tests)...